pkg/validators/am0004_namespace.go

```diff
@@ -0,0 +1,88 @@
+package validators
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/utils"
+)
+
+var namespaceRegexExceptions = []string{
+	"acm",
+	"codeready-workspaces-operator",
+	"codeready-workspaces-operator-qe",
+	"addon-dba-operator",
+	"openshift-storage",
+	"prow",
+}
+
+var namespacePresenceExceptions = []string{
+	"openshift-logging",
+}
+
+var namespaceRegex = regexp.MustCompile(`^redhat-.*$`)
+
+var AM0004 = utils.Validator{
+	Code:        "AM0004",
+	Name:        "ensure_namespace",
+	Description: "Ensure that the target namespace is listed in the set of channels listed",
+	Runner:      ValidateNamespace,
+}
+
+func init() {
+	Registry.Add(AM0004)
+}
+
+func ValidateNamespace(metabundle utils.MetaBundle) (bool, string, error) {
+	targetNamespace := metabundle.AddonMeta.TargetNamespace
+	namespaceList := metabundle.AddonMeta.Namespaces
+	valid := validateNamespacePresence(targetNamespace, namespaceList, namespacePresenceExceptions)
+	if !valid {
+		return false, "Target namespace is not in the list of supplied namespaces", nil
+	}
+
+	allValid, failedNamespaces := validateNamespaceRegex(namespaceList, namespaceRegexExceptions)
+	if !allValid {
+		return false, fmt.Sprintf("Some namespaces doesn't start with 'redhat-*' %v", failedNamespaces), nil
+	}
+	return true, "", nil
+}
+
+func validateNamespaceRegex(namespaceList []string, exceptionList []string) (bool, []string) {
+	failedNamespaces := []string{}
+	for _, namespace := range namespaceList {
+		if includes(namespace, exceptionList) {
+			continue
+		}
+		matched := namespaceRegex.MatchString(namespace)
+		if !matched {
+			failedNamespaces = append(failedNamespaces, namespace)
+		}
+	}
+	if len(failedNamespaces) == 0 {
+		return true, []string{}
+	}
+	return false, failedNamespaces
+}
+
+func validateNamespacePresence(targetNamespace string, namespaceList []string, exceptionList []string) bool {
+	if includes(targetNamespace, exceptionList) {
+		return true
+	}
+
+	for _, namespace := range namespaceList {
+		if namespace == targetNamespace {
+			return true
+		}
+	}
+	return false
```

Comment:
```suggestion
	return includes(targetNamespace, exceptionList) || includes(targetNamespace, namespaceList)
```

Reply:
@yashvardhan-kukreja  I think we can keep the function, it adds more clarity to someone whose reading the code later. 
I think `return includes(targetNamespace, exceptionList) || includes(targetNamespace, namespaceList)` this refactor is a little cryptic.
I think this is more readable, Wdyt?:
```
if includes(targetNamespace, exceptionList) {
     return true
}

if includes(targetNamespace, namespaceList){
     return true
}
return false
```

Reply:
I am a big proponent of cleaner code so I might be having a personal bias towards `return includes(targetNamespace, exceptionList) || includes(targetNamespace, namespaceList)`
or
```
// If targetNamespace is in the exceptionlist, return true
if includes(targetNamespace, exceptionList) {
     return true
}
// If targetNamespace is in the namespaceList, return true
returns includes(targetNamespace, namespaceList)
```

But yeah, it's not a hard requirement from my end coz it's not creating any performance-oriented difference.
So, feel free to move ahead with anything and resolve this thread :)
Thanks!

Reply:
@yashvardhan-kukreja 
Lets do `includes(targetNamespace, exceptionList) || includes(targetNamespace, namespaceList)`  :)